### PR TITLE
Delay listening until readiness checks have succeeded.

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+## 2.8.0 - TBD
+
+### Added
+- Do not listen for HTTP/HTTPS requests until readiness checks have passed. This
+  is compatible with deployment platforms that lack explicit health and
+  readiness checks, but instead expect that the application is ready when it
+  responds to requests sent to the HTTP/HTTPS server's TCP port.
+  - Emit a new `bedrock-server.readiness-check` event.
+  - Delay the start of the HTTP/HTTPS servers until the
+    `bedrock-server.readiness-check` listners have succeeded.
```
Comment (dlongley) [inline CHANGELOG.md:12]:
```suggestion
    `bedrock-server.readiness-check` listeners have succeeded.
```

**Review by mandyvenables — APPROVED** [review]

**Review by dlongley — APPROVED** [review]
Thanks! This seems ok at a glance. I don't know what kind of deadlocks it may (or may not) introduce -- if that becomes a problem hopefully it can be remedied with documentation and different specific solutions to avoid it.

This change also made me wonder what our naming convention is for event names -- whether we're using hyphens for local event names or just for the module/namespace names that precede them. I'm willing to accept the name as-is, but if there were a single name that was simpler that could replace `readiness-check` that would be preferable. I know we use `camelCase` for bedrock-express event names here:

https://github.com/digitalbazaar/bedrock-express/blob/main/lib/index.js#L133-L161

So if we do go with two words, perhaps `readinessCheck` is more inline with precedent.

File: CHANGELOG.md (hunk 1 above)
Comment (mattcollier) [reply CHANGELOG.md]: TY, fixed.

**Comment by mattcollier** [comment]
Changed to `readinessCheck` to match convention.

**Pull request merged**